python/lsst/cp/pipe/cpCombine.py

```diff
@@ -860,3 +887,158 @@ def setFlatSource(self, exp):
             return
 
         exp.metadata["FLATSRC"] = self.config.flatSource
+
+
+class CalibCombineTwoFlatsByFilterConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("instrument", "detector", "physical_filter"),
+):
+    inputFlatOneHandle = cT.Input(
+        name="flat_blue",
+        doc="Input first processed flat to combine.",
+        storageClass="Exposure",
+        dimensions=("instrument", "detector", "physical_filter"),
+        multiple=False,
+        deferLoad=True,
+        isCalibration=True,
+    )
+    inputFlatTwoHandle = cT.Input(
+        name="flat_red",
+        doc="Input second processed flat to combine.",
+        storageClass="Exposure",
+        dimensions=("instrument", "detector", "physical_filter"),
+        multiple=False,
+        deferLoad=True,
+        isCalibration=True,
+    )
+    outputData = cT.Output(
+        name="flat",
+        doc="Output combined flat.",
+        storageClass="ExposureF",
+        dimensions=("instrument", "detector", "physical_filter"),
+        isCalibration=True,
+    )
+
+
+class CalibCombineTwoFlatsByFilterConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=CalibCombineTwoFlatsByFilterConnections,
+):
+    calibrationType = pexConfig.Field(
+        dtype=str,
+        default="flat",
+        doc="Name of calibration to be generated.",
+    )
+    weightOne = pexConfig.Field(
+        dtype=float,
+        default=0.5,
+        doc="Weight for first flat (second weight will be 1.0 - weightOne).",
+    )
+    noGoodPixelsMask = pexConfig.Field(
+        dtype=str,
+        default="BAD",
+        doc="Mask bit to set when there are no good input pixels.  See code comments for details.",
+    )
+    combine = pexConfig.Field(
+        dtype=str,
+        default="MEAN",
+        doc="Statistic name to use for combination (from `~lsst.afw.math.Property`)",
+    )
+    subregionSize = pexConfig.ListField(
+        dtype=int,
+        doc="Width, height of subregion size.",
+        length=2,
+        # This is 200 rows for all detectors smaller than 10k in width.
+        default=(10000, 200),
+    )
+    distributionPercentiles = pexConfig.ListField(
+        dtype=float,
+        default=[0, 5, 16, 50, 84, 95, 100],
+        doc="Percentile levels to measure on the final combined calibration.",
+    )
+
+
+class CalibCombineTwoFlatsByFilterTask(CalibCombineTask):
```

Comment:
I think I'm fine with this, but if the scaling was applied at the `cpFlatMeasure` step, then we could just use the standard `cpCombine`.  I was also thinking about an `SED-model` that would hold a set of weights for a given filter that would make this extendable to arbitrary "monochromatic" inputs.  Maybe that's more complicated than it needs to be for the two-input case.

Reply:
I didn't want to get too far down the monochromatic line since I don't know how we're going to do that and don't want to guess.  But as commented on Jira due to the different gradients I think we're going to need to insert a gradient-removal step here.